From 04e1bca5ec022dcec12044b866f3e7bb4378c7ff Mon Sep 17 00:00:00 2001
From: Friedemann Kleint <Friedemann.Kleint@qt.io>
Date: Wed, 26 Feb 2025 16:14:50 +0100
Subject: [PATCH] Fix PySide build for 6.10

Task-number: PYSIDE-3011
Task-number: QTBUG-87776
Change-Id: If1bbeb6599ef60d72f49a949a30b01f9140c8d97
Reviewed-by: Ece Cinucen <ece.cinucen@qt.io>
Reviewed-by: Cristian Maureira-Fredes <cristian.maureira-fredes@qt.io>
(cherry picked from commit e5b3dcfe685bf250d25f48111db289c175b112cc)
---
 sources/pyside6/CMakeLists.txt            | 2 ++
 sources/pyside6/cmake/PySideHelpers.cmake | 7 ++++---
 sources/pyside6/cmake/PySideSetup.cmake   | 3 ++-
 3 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/sources/pyside6/CMakeLists.txt b/sources/pyside6/CMakeLists.txt
index 9c238e980..423b96b6f 100644
--- a/sources/pyside6/CMakeLists.txt
+++ b/sources/pyside6/CMakeLists.txt
@@ -9,6 +9,8 @@ cmake_policy(VERSION 3.18)
 include(".cmake.conf")
 project(pysidebindings)
 
+set(QT_NO_PRIVATE_MODULE_WARNING ON)
+
 include(cmake/PySideSetup.cmake)
 
 get_rpath_base_token(base)
diff --git a/sources/pyside6/cmake/PySideHelpers.cmake b/sources/pyside6/cmake/PySideHelpers.cmake
index a4641ee65..0833cdd2d 100644
--- a/sources/pyside6/cmake/PySideHelpers.cmake
+++ b/sources/pyside6/cmake/PySideHelpers.cmake
@@ -255,9 +255,10 @@ macro(collect_module_if_found shortname)
         # record the shortnames for the tests
         list(APPEND all_module_shortnames ${shortname})
         # Build Qt 5 compatibility variables
-        if(${QT_MAJOR_VERSION} GREATER_EQUAL 6 AND NOT "${shortname}" STREQUAL "OpenGLFunctions")
-            get_target_property(Qt6${shortname}_INCLUDE_DIRS Qt6::${shortname}
-                                INTERFACE_INCLUDE_DIRECTORIES)
+        get_target_property(Qt6${shortname}_INCLUDE_DIRS Qt6::${shortname}
+                            INTERFACE_INCLUDE_DIRECTORIES)
+        # Find QtGui private headers for exposing some QPA classes
+        if("${shortname}" STREQUAL "Gui")
             get_target_property(Qt6${shortname}_PRIVATE_INCLUDE_DIRS
                                 Qt6::${shortname}Private
                                 INTERFACE_INCLUDE_DIRECTORIES)
diff --git a/sources/pyside6/cmake/PySideSetup.cmake b/sources/pyside6/cmake/PySideSetup.cmake
index 45a63a1a0..444f33ed9 100644
--- a/sources/pyside6/cmake/PySideSetup.cmake
+++ b/sources/pyside6/cmake/PySideSetup.cmake
@@ -97,7 +97,8 @@ if(QFP_QT_HOST_PATH)
         endif()
     endif()
 endif()
-find_package(Qt6 REQUIRED COMPONENTS Core)
+# Find QtGui private headers for exposing some QPA classes
+find_package(Qt6 REQUIRED COMPONENTS Core CorePrivate Gui GuiPrivate)
 
 add_definitions(${Qt${QT_MAJOR_VERSION}Core_DEFINITIONS})
 
-- 
2.51.2

